src/Microsoft.ApplicationInsights/Managed/Shared/Extensibility/Implementation/Tracing/HeartbeatProvider.cs

```diff
@@ -0,0 +1,328 @@
+
+        public void Dispose()
+        {
+            this.Dispose(true);
+        }
+
+        /// <summary>
+        /// Get the metric telemetry item that will be sent.
+        /// 
+        /// Note: exposed to internal to allow inspection for testing.
+        /// </summary>
+        /// <returns>A MetricTelemtry item that contains the currently defined payload for a heartbeat 'pulse'.</returns>
+        internal ITelemetry GatherData()
+        {
+            var hbeat = new MetricTelemetry(heartbeatSyntheticMetricName, 0.0);
+
+            string updatedKeys = string.Empty;
+            string comma = string.Empty;
+
+            if (hbeat.Properties.ContainsKey(HeartbeatDefaultPayload.UpdatedFieldsPropertyKey))
+            {
+                updatedKeys = hbeat.Properties[HeartbeatDefaultPayload.UpdatedFieldsPropertyKey];
+                comma = ",";
+            }
+
+            foreach (var payloadItem in this.heartbeatProperties)
+            {
+                hbeat.Properties.Add(payloadItem.Key, payloadItem.Value.PayloadValue);
+                hbeat.Sum += payloadItem.Value.IsHealthy ? 0 : 1;
+                if (payloadItem.Value.IsUpdated)
+                {
+                    string.Concat(updatedKeys, comma, payloadItem.Key);
+                    comma = ",";
+                }
+            }
+
+            // update the special 'updated keys' property with the names of keys that have been updated.
+            if (!string.IsNullOrEmpty(updatedKeys))
+            {
+                hbeat.Properties[HeartbeatDefaultPayload.UpdatedFieldsPropertyKey] = updatedKeys;
+            }
+
+            hbeat.Sequence = string.Format(CultureInfo.CurrentCulture, "{0}", this.heartbeatsSent++);
+
+            return hbeat;
+        }
+
+        protected void Dispose(bool disposing)
+        {
+            if (!this.disposedValue)
+            {
+                if (disposing)
+                {
+                    if (this.HeartbeatTimer != null)
+                    {
+                        this.isEnabled = false;
+
+                        try
+                        {
+                            this.HeartbeatTimer.Dispose();
+                        }
+                        catch (Exception)
+                        {
+                            CoreEventSource.Log.LogError("Error occured when disposing heartbeat timer within HeartbeatProvider");
+                        }
+                    }
+                }
+
+                this.disposedValue = true;
+            }
+        }
+
+        #endregion
+
+        private void Send()
+        {
+            if (this.telemetryClient.TelemetryConfiguration.TelemetryChannel == null)
+            {
+                return;
+            }
+
+            var eventData = (MetricTelemetry)this.GatherData();
+
+            eventData.Context.Operation.SyntheticSource = heartbeatSyntheticMetricName;
+
+            eventData.Context.InstrumentationKey = this.InstrumentationKey;
```

Comment:
you already set it on telemetry client. No need to set on individual item now

Reply:
Ok.